src/main/java/org/broadinstitute/dsm/jobs/TestBostonUPSTrackingJob.java

```diff
@@ -58,36 +59,71 @@ public void execute(JobExecutionContext context) throws JobExecutionException {
+                                        rs.getString("kit." + DBConstants.DSM_RECEIVE_DATE),
+                                        rs.getString("kit." + DBConstants.UPS_TRACKING_STATUS),
+                                        rs.getString("kit." + DBConstants.UPS_TRACKING_DATE),
+                                        rs.getString("kit." + DBConstants.UPS_RETURN_STATUS),
+                                        rs.getString("kit." + DBConstants.UPS_RETURN_DATE),
+                                        rs.getString("req." + DBConstants.COLLABORATOR_PARTICIPANT_ID),
+                                        rs.getString("req." + DBConstants.EXTERNAL_ORDER_NUMBER),
+                                        rs.getBoolean("kit." + DBConstants.CE_ORDER)
+                                );
+                                String type;
+                                if (StringUtils.isNotBlank(kit.getTrackingToId())) {
+                                    type = kit.getUpsTrackingStatus();
+                                    if (StringUtils.isNotBlank(type)) {// get only type from it
+                                        type = type.substring(0, type.indexOf(' '));
+                                    }
+                                    if (!"D".equals(type)) {//don't include delivered ones
+                                        try {
+                                            updateKitStatus(conn, kit, false);
+                                        } catch (Exception e) {
+                                            logger.error("Could not update outbound status for " + kit.getExternalOrderNumber(), e);
+                                        }
+                                    }
                                 }
-                            }
-                        }
-                        kits = ids.get("return");
-                        if (kits != null) {
-                            logger.info("checking return status for " + kits.size() + " tracking numbers");
-                            for (DdpKit kit : kits.values()) {
-                                try {
-                                    updateKitStatus(kit, true);
-                                } catch (Exception e) {
-                                    logger.error("Could not update return status for " + kit.getExternalOrderNumber(), e);
+                                if (StringUtils.isNotBlank(kit.getTrackingReturnId())) {
+                                    type = kit.getUpsReturnStatus();
+                                    if (StringUtils.isNotBlank(type)) {
+                                        type = type.substring(0, type.indexOf(' '));
+                                    }
+                                    if (!"D".equals(type)) {
+                                        if (kit.getKitLabel().contains("_1") && kit.getKitLabel().indexOf("_1") == kit.getKitLabel().length() - 2) {
```

Comment:
Magic string alert...what is `"_1"`, and what does it mean business-wise?  Instead of dipping into the label and doing an a check on it, write a method that tells you what you need to know.

Reply:
@pegahtah can you double check why you checked if label has `_1` 
if it is `_1` you remove it and update the label to substring without `_1`